lib/std/os/linux/io_uring.zig

```diff
@@ -522,6 +541,25 @@ pub const IO_Uring = struct {
         return sqe;
     }
 
+    /// Queues (but does not submit) an SQE to perform a `recv(2)`.
+    /// Returns a pointer to the SQE.
+    ///
+    /// The only difference with `recv` is that this takes a nullable many-item pointer and a length separately.
+    /// This lets the caller pass null as the buffer, which is required when using the automatic buffer selection.
+    pub fn recv_raw(
+        self: *IO_Uring,
+        user_data: u64,
+        fd: os.fd_t,
+        buffer: ?[*]u8,
+        len: usize,
+        flags: u32,
+    ) !*io_uring_sqe {
```

Comment:
```suggestion
    pub fn recv_with_automatic_buffer_selection(
        self: *IO_Uring,
        user_data: u64,
        fd: os.fd_t,
        group_id: usize,
        flags: u32,
    ) !*io_uring_sqe {
```

@vrischmann What do you think if this interface is explicitly only used for automatic buffer selection, not for normal recv/read calls where buffers are provided, since passing by slice there provides more safety in those cases?

This could then also set the `IOSQE_BUFFER_SELECT` in the prep routine, to ensure there's no confusion about where this should be done. I know this is different to how liburing does it, but I think that makes sense for C, and perhaps not Zig, which has richer types.

This interface was the one thing I wasn't sure about when I did the original io_uring PR.

Reply:
Another idea for consideration: have read/recv take a `Buffer` type which is
```zig
union(enum) {
  bytes: []u8,
  iovecs: []const os.iovec,
  buffer_group_id: usize,
}
```
The variant would then call the correct prep() function. Not sure if this is any better than just having separate functions.

Reply:
Thanks for your feedback.

I like the idea of using a union for the buffer type, I was hesitant to do this because it will break all existing code using io_uring. If it's acceptable I'll get back into this PR and look at implementing it.